### ISSUE TITLE
PA3011: Roundtrip validation on unpack failed.  Version 1.8.6

### ISSUE THREAD

**Issue by EPTS-Dev** [microsoft/PowerApps-Language-Tooling#296]
UnPack from C:\PowerApps\MSAppFiles\OP+IntakeSprint40FullBaseline.msapp-->C:\PowerApps\ProjectFolders\Sprint40Full
  source format version: 0.24                                                                                          C:\PowerApps\ProjectFolders\Sprint40Full
Error   PA3011: Roundtrip validation on unpack failed. This is always a bug, please file an issue at https://github.com/microsoft/PowerApps-Language-Tooling
Error   PA3011: Roundtrip validation on unpack failed. This is always a bug, please file an issue at https://github.com/microsoft/PowerApps-Language-Tooling/microsoft/PowerApps-Language-Tooling                                                                                  /microsoft/PowerApps-Language-Tooling
Microsoft PowerPlatform CLI
Version: 1.8.6+g6a1721e

Contact me if you need more information.

**Comment by lesaltzm** [comment]
Hey @EPTS-Dev, thanks for the report. Was that the only message you saw when running the unpack command? Could you please send a copy of the app to me (lesaltzm at microsoft.com) so I can take a look and see what’s going wrong here.

**Comment by MikeStall** [comment]
This could be the same issue as #282

**Comment by EPTS-Dev** [comment]
Yes it could but there is no way of knowing for sure expect that it is the same error code.  The version where it appears is different though 1.6 to 1.8 so logged as a new issue so it doesn't get overlooked.

**Comment by MikeStall** [comment]
@EPTS-Dev  - can you send the msapp?  fwiw, It looks like #282 is because of an older app.

**Comment by MikeStall** [comment]
Looking .. the difference here is in the TestStepsMetadata property.

**Comment by MikeStall** [comment]
There's a \n  in the test's Description field. It's getting roundtripped as \r\n. 
There's significant extra processing that happens for TestStepsMetadata (see https://github.com/microsoft/PowerApps-Language-Tooling/blob/master/src/PAModel/SourceTransforms/AppTestTransform.cs ), so it's probably in there. 

As a workaround, try removing the \n at the end of the property (will send email on which one)

**Comment by MikeStall** [comment]
(updating title)

**Comment by EPTS-Dev** [comment]
> There's a \n in the test's Description field. It's getting roundtripped as \r\n.
> There's significant extra processing that happens for TestStepsMetadata (see https://github.com/microsoft/PowerApps-Language-Tooling/blob/master/src/PAModel/SourceTransforms/AppTestTransform.cs ), so it's probably in there.
> 
> As a workaround, try removing the \n at the end of the property (will send email on which one)

Hi Mike 

The workaround did not work for me.  Removing the /n (new line) from any Test Studio TestStep, then saving the app again to the desktop and trying to import in VS Code with the PowerApps CLI still generated a roundtrip validation error.  I even went as far as to remove the teststeps and I still get the error.

**Comment by EPTS-Dev** [comment]
I removed all new lines from the test steps and the test on start, on completion, etc areas.  I am still getting the Error   PA3011: Roundtrip validation on unpack failed errors (2 instances)  If you want to see the .msapp file let me know where to send it.

**Comment by MikeStall** [comment]
Send it to jmstall at microsoft dot com.  Thanks.

**Comment by NavneetThekkumpat** [comment]
@EPTS-Dev :
As another workaround, please try closing quotes with the description and not on new line (on all the test steps and actions). 
Also try removing all extra spaces or empty new lines too.

**Comment by landonarogers** [comment]
@EPTS-Dev Do you have a current way we could reproduce this issue in PowerApps, so that we can see on our end what exactly causes the error? Thank you for your time and patience!
